aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1851,6 +1815,89 @@ static Tensor& linalg_norm_out_impl(Tensor& result, const Tensor& self, optional
   return result;
 }
 
+static Tensor& linalg_vector_norm_impl(const Tensor& self, optional<Scalar> opt_ord, optional<IntArrayRef> opt_dim, bool keepdim, optional<ScalarType> opt_dtype, Tensor& result) {
+  // Casting a large integer to a double will introduce some error, but for
+  // practical purposes, it won't matter since a large order will usually
+  // give an infinite result
+  auto ord = opt_ord.value_or(2).toDouble();
+
+  TORCH_CHECK(self.device().type() == DeviceType::CPU || self.device().type() == DeviceType::CUDA,
+              "vector_norm only supports CPU and CUDA device types, but got: ", self.device().type());
+  TORCH_CHECK(self.layout() == Layout::Strided,
+              "vector_norm only supports strided layout, but got: ", self.layout());
+
+  if (opt_dtype.has_value() && isComplexType(self.scalar_type())) {
+    TORCH_CHECK(isComplexType(opt_dtype.value()),
+      "vector_norm expected complex 'dtype', since input is complex, ",
+      "but got ", opt_dtype.value());
+  }
+
+  ScalarType in_dtype = opt_dtype.value_or(self.scalar_type());
+  TORCH_CHECK(
+      at::isFloatingType(in_dtype) || at::isComplexType(in_dtype),
+      "vector_norm only supports floating point and complex dtypes, but got: ",
+      toString(in_dtype));
+
+  IntArrayRef dim = opt_dim.value_or(IntArrayRef{});
+
+  if (self.numel() == 0) {
+    // TODO: The question about how to handle negative orders when the input
+    // is empty has not been settled yet. For now, we raise an error. Issue:
+    // https://github.com/pytorch/pytorch/issues/52783
+    TORCH_CHECK(ord >= 0,
+      "vector_norm of negative order cannot be performed on an empty tensor");
+
+    // For NumPy compatibility, we can only perform order infinity reduction
+    // (max/min) on a tensor with zero elements if the dimensions to reduce are
+    // nonzero. Otherwise, throw an error.
+    if (ord == INFINITY) {
+      bool has_identity = true;
+
+      if (dim.size() == 0) {
+        has_identity = false;
+      } else {
+        for (int64_t dim_num : dim) {
+          if (self.size(dim_num) == 0) {
+            has_identity = false;
+            break;
+          }
+        }
+      }
+      TORCH_CHECK(has_identity,
+        "vector_norm cannot compute the infinity norm on an empty tensor ",
```

Comment:
this error message seems misleading because the infinity norm can be computed on an empty tensor, it just can't be computed on a dimension of size zero

Reply:
Updated to say "empty dimension" instead of "empty tensor"